### PR TITLE
onboard vsi extension to catalog

### PR DESCRIPTION
**Pull request by ocofaigh**
### Description

onboard vsi extension to catalog

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by ocofaigh** [comment]
Still work to do in this PR, but @SirSpidey looking for early feedback around naming here

**Comment by SirSpidey** [comment]
@ocofaigh some ideas based on how I assume this will display to users

### variation name

**Original**
- "label": "VSI on existing VPC"

**Suggested**
- "label": "Existing VPC" 

**Alternatives**
- Extend VPC with VSI
- Add VSI to VPC

#### Catalog
image


#### Deployment guide
image

**Comment by ocofaigh** [comment]
Thanks @SirSpidey I have updated to "Existing VPC"

**Comment by SirSpidey** [comment]
@ocofaigh Do we want more feedback on this first extension? (To find out whether the label is clear.) Of course, we should add details in the description under _Learn more_. (And in the deployment guide.)

**Comment by SirSpidey** [comment]
@ocofaigh Keith confirms that the dropdown mockup is the way this will get displayed. So I guess the name is irrelevant. The UI will display `Extend VSI on VPC landing zone` in the dropdown. When you click **Extend**, only the standard variation/flavor tile will remain, assuming that’s the prereq (quickstart tile will disappear).

**Comment by ocofaigh** [comment]
More updates needed in this PR.. working on them now..

File: ibm_catalog.json [hunk 1]
```diff
@@ -1,1271 +1,1365 @@
...
+          "architecture": {
+            "descriptions": "This architecture creates virtual server instances (VSI) for VPC in some or all of the subnets of any existing landing zone VPC deployable architecture.\r\n\r\n## Before you begin\r\n\r\n- You must have either the [VPC landing zone](https:\/\/cloud.ibm.com\/catalog\/architecture\/deploy-arch-ibm-slz-vpc-9fc0fa64-27af-4fed-9dce-47b3640ba739-global) or [Red Hat OpenShift Container Platform on VPC landing zone](https:\/\/cloud.ibm.com\/catalog\/architecture\/deploy-arch-ibm-slz-ocp-95fccffc-ae3b-42df-b6d9-80be5914d852-global) deployable architecture deployed.\r\n- The block storage to KMS auth policy must exist. This policy would have been created by one of the above deployable architectures if the `add_kms_block_storage_s2s` variable was set to `true`, which is default value.\r\n- You need the VPC ID, subnet names, and boot volume encryption key from your existing landing zone VPC deployable architecture. For information about finding these values, see [Adding a VSI to your VPC landing zone deployable architecture](https:\/\/cloud.ibm.com\/docs\/secure-infrastructure-vpc?topic=secure-infrastructure-vpc-ext-with-vsi).",
```
Comment (SirSpidey) [inline ibm_catalog.json:555]: see notes in review comments

File: ibm_catalog.json [hunk 2]
```diff
@@ -1,1271 +1,1365 @@
...
+            "features": [
+              {
+                "title": "Virtual Server Instances for every subnet",
```
Comment (SirSpidey) [inline ibm_catalog.json:558]:
```suggestion
                "title": "Adds a virtual server instance in every VPC subnet",
```

File: ibm_catalog.json [hunk 3]
```diff
@@ -1,1271 +1,1365 @@
...
+              {
+                "title": "Increases security with Key Management",
+                "description": "Yes"
+              },
+              {
+                "title": "Simplifies risk management and demonstrates regulatory compliance with Financial Services",
+                "description": "Yes"
+              },
+              {
+                "title": "Uses Floating IP address for access through the public internet",
```
Comment (SirSpidey) [inline ibm_catalog.json:570]:
```suggestion
                "title": "Uses floating IP addresses for access through the public internet",
```

File: ibm_catalog.json [hunk 4]
```diff
@@ -1,1271 +1,1365 @@
...
-                    }
-                }
+            "diagrams": [
+              {
+                "diagram": {
+                  "caption": "VSI on existing VPC",
+                  "url": "https://raw.githubusercontent.com/terraform-ibm-modules/terraform-ibm-landing-zone/main/reference-architectures/vsi-vsi.drawio.svg",
+                  "type": "image/svg+xml"
+                },
+                "description": "The VSI on existing VPC deployable architecture extension is based on the IBM Cloud for Financial Services reference architecture. The architecture creates a customizable and secure infrastructure, with virtual servers, to run your workloads with a Virtual Private Cloud (VPC) in multizone regions."
```
Comment (SirSpidey) [inline ibm_catalog.json:581]:
```suggestion
                "description": "The extension to an existing VPC deployable architecture is based on the IBM Cloud for Financial Services reference architecture. The architecture creates a customizable and secure infrastructure with virtual servers to run your workloads with IBM Cloud VPC in multizone regions."
```

**Review by SirSpidey — CHANGES_REQUESTED** [review]
Reviewed only from line 499

I think we need to modify the overview in both this json and extension readme file. 

The statements here assume that you have already deployed a DA, which I think is the workflow.

`The block storage to KMS auth policy must exist. This policy would have been created by one of the above deployable architectures if the add_kms_block_storage_s2s variable was set to true, which is default value.`

Suggestion
- You need an authorization policy that grants access between block storage and the KMS. The policy exists if you set the `add_kms_block_storage_s2s` input variable to `true` (the default value) in your existing landing zone deployable architecture.

---

`You need the VPC ID, subnet names, and boot volume encryption key from your existing landing zone VPC deployable architecture. For information about finding these values, see [Adding a VSI to your VPC landing zone deployable architecture](https://cloud.ibm.com/docs/secure-infrastructure-vpc?topic=secure-infrastructure-vpc-ext-with-vsi).`

Suggestion
- You need the VPC ID, subnet names, and boot volume encryption key from your existing landing zone deployable architecture. For information about finding these values, see [Adding a VSI to your VPC landing zone deployable architecture](https://cloud.ibm.com/docs/secure-infrastructure-vpc?topic=secure-infrastructure-vpc-ext-with-vsi).

**Comment by ocofaigh** [comment]
Thanks for the feedback @SirSpidey - I robbed that text from other variations in this json already so Ill have to update those too to be consistent

**Comment by SirSpidey** [comment]
> Thanks for the feedback @SirSpidey - I robbed that text from other variations in this json already so Ill have to update those too to be consistent

@ocofaigh Feel free to delegate that to me.

**Comment by ocofaigh** [comment]
@SirSpidey I've made the changes locally. Working on other changes now so will be included in my next commit

File: patterns/vsi-extension/README.md [hunk 5]
```diff
@@ -2,10 +2,12 @@
 
 This architecture creates virtual server instances (VSI) for VPC in some or all of the subnets of any existing landing zone VPC deployable architecture.
 
+NOTE: This solution only supports creating a VSI in a single VPC. If you wish to create VSIs in multiple VPCs you must create one instance of the solution per VPC.
+
```
Comment (SirSpidey) [inline patterns/vsi-extension/README.md:6]:
```suggestion
This architecture creates virtual server instances (VSI) in some or all of the subnets of one VPC of an existing landing zone deployable architecture. To create VSIs in multiple VPCs, deploy the extension once for each VPC.
```
suggestion to pull it all together

**Comment by ocofaigh** [comment]
The prereq script is working well now. I was able to import a version to catalog, however its not displaying. Some issue with the extension type. The catalog team are debugging now

**Comment by ocofaigh** [comment]
Need to wait until https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone/pull/648 is merged first and then I'll rebase this PR and kick off pipeline

**Review by Aashiq-J — APPROVED** [review]
LGTM

**Pull request merged**